In [5]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
from scipy import stats
from scipy.stats import chi2_contingency,chisquare
import warnings
warnings.filterwarnings('ignore')

steps for hypothesis testing

1. define null and alternate hypothesis

2.define which test to be performed z or t test

3.define significant value i.e. alpha value (normally 0.05)

4.find p value from t value

5.compare p value with alpha

  i. if alpha < P value
         reject null hypoyhesis
  ii.if alpha > p value
          fail to reject null hypothesis

Q1) A F&B manager wants to determine whether there is any significant difference in the diameter of the cutlet between two units. A randomly selected sample of cutlets was collected from both units and measured? Analyze the data and draw inferences at 5% significance level. Please state the assumptions and tests that you carried out to check validity of the assumptions.

In [8]:
# importing and defining the dataset
df1=pd.read_csv('Cutlets.csv')
# seeing the first 5 rows of the dataset
df1[:5]

Unit A  Unit B
0  6.8090  6.7703
1  6.4376  7.5093
2  6.9157  6.7300
3  7.3012  6.7878
4  7.4488  7.1522

EDA



In [9]:
# to see the information about the dataset
df1.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 35 entries, 0 to 34
Data columns (total 2 columns):
 #   Column  Non-Null Count  Dtype  
---  ------  --------------  -----  
 0   Unit A  35 non-null     float64
 1   Unit B  35 non-null     float64
dtypes: float64(2)
memory usage: 688.0 bytes


In [10]:
# to see the statistical description about the dataset
df1.describe()

Unit A     Unit B
count  35.000000  35.000000
mean    7.019091   6.964297
std     0.288408   0.343401
min     6.437600   6.038000
25%     6.831500   6.753600
50%     6.943800   6.939900
75%     7.280550   7.195000
max     7.516900   7.545900

In [11]:
# to see the sum of any duplicated values of the dataset
df1.duplicated().sum()

0

In [12]:
# to see the unique values in the dataset
df1.value_counts()



Unit A  Unit B
6.4376  7.5093    1
7.2828  7.3070    1
7.0852  7.5459    1
7.1560  7.4220    1
7.1561  6.6965    1
7.2163  7.0133    1
7.2705  7.4314    1
7.2783  7.1180    1
7.2854  7.1688    1
6.9952  6.7594    1
7.3012  6.7878    1
7.3495  6.7478    1
7.3871  6.8110    1
7.3930  6.8810    1
7.3943  6.5780    1
7.4488  7.1522    1
7.0621  6.6606    1
6.9438  6.8889    1
6.5341  6.5217    1
6.8236  7.0503    1
6.5797  7.1581    1
6.5970  6.6672    1
6.6801  6.9182    1
6.6840  7.2402    1
6.7794  7.0992    1
6.8090  6.7703    1
6.8394  7.0240    1
6.9431  6.3346    1
6.8568  6.9399    1
6.8755  7.2212    1
6.9157  6.7300    1
6.9246  6.7652    1
6.9256  6.0380    1
6.9405  7.3875    1
7.5169  7.4059    1
dtype: int64

defining the hypothesis

here we can see that for null hypothesis, UnitA = UnitB & for alternate hypothesis UnitA != UnitB



In [13]:
# predicted value by t test
p_value1=stats.ttest_ind(df1['Unit A'],df1['Unit B'],alternative='two-sided')[1]
p_value1

0.4722394724599501

In [14]:
if p_value1 < 0.05:
    print("H0 is rejected and there is no significant difference in the diameter of cutlet between two units")
else:
    print("H0 is accepted and there is a significant difference in the diameter of cutlet between two units")

H0 is accepted and there is a significant difference in the diameter of cutlet between two units


**from the above we can see H0=47.22% and H1=100-47.22=52.88% for 5% signifacant level , here p value is greater than alpha so 'REJECT THE NULL HYPOTHESIS'**

**Means there is a significant difference in the diameter of the cutlet between two units.**

# Q2) A hospital wants to determine whether there is any difference in the average Turn Around Time (TAT) of reports of the laboratories on their preferred list. They collected a random sample and recorded TAT for reports of 4 laboratories. TAT is defined as sample collected to report dispatch.
# Analyze the data and determine whether there is any difference in average TAT among the different laboratories at 5% significance level. **bold text**

In [15]:
# importing the dataset
df2=pd.read_csv('LabTAT.csv')
df2

Laboratory 1  Laboratory 2  Laboratory 3  Laboratory 4
0          185.35        165.53        176.70        166.13
1          170.49        185.91        198.45        160.79
2          192.77        194.92        201.23        185.18
3          177.33        183.00        199.61        176.42
4          193.41        169.57        204.63        152.60
..            ...           ...           ...           ...
115        178.49        170.66        193.80        172.68
116        176.08        183.98        215.25        177.64
117        202.48        174.54        203.99        170.27
118        182.40        197.18        194.52        150.87
119        182.09        215.17        221.49        162.21

[120 rows x 4 columns]

for null hypothesis: lab1=lab2=lab3=lab4

for alternate hypothesis: lab1!=lab2!=lab3!=lab4



here 4 different data so we would use here Anova test

In [16]:
# to see the pvalues by f_oneway method
p_value2=stats.f_oneway(df2.iloc[:,0],df2.iloc[:,1],df2.iloc[:,2],df2.iloc[:,3])[1]
p_value2

2.1156708949992414e-57

In [17]:
if p_value2 < 0.05:
    print('reject the null hypothesis and  there is not difference in average TAT among the different laboratories ')
else:
    print('failed to reject the null hypothesis and  there is a difference in average TAT among the different laboratories')

reject the null hypothesis and  there is not difference in average TAT among the different laboratories 


# here we can see that p values is less than alpha i.e. 0.05, so we reject the null hypothesis means "there is difference in average TAT among the different laboratories"

# Q3) Sales of products in four different regions is tabulated for males and females. Find if male-female buyer rations are similar across regions.

In [18]:
# importing the dataset
df3=pd.read_csv('BuyerRatio.csv')
df3

Observed Values  East  West  North  South
0           Males    50   142    131     70
1         Females   435  1523   1356    750

In [19]:
#here we can see that there are two categories data so here we need to use chi square test

In [20]:
# defining the array by removing the serial number column
df3_1=np.array(df3.iloc[:,1:5])

In [21]:
# p value by chi square contingency test
p_value3=chi2_contingency(df3_1)[1]
p_value3

0.6603094907091882

In [22]:
if p_value3 < 0.05:
    print("H0 is rejected and there is no Male and female buyers are similar across in region")
else:
    print("H0 is accepted and Male and female buyers are similar across in region")

H0 is accepted and Male and female buyers are similar across in region


# Q4) TeleCall uses 4 centers around the globe to process customer order forms. They audit a certain % of the customer order forms. Any error in order form renders it defective and has to be reworked before processing. The manager wants to check whether the defective % varies by centre. Please analyze the data at 5% significance level and help the manager draw appropriate inferences

In [23]:
# importing the dataset
df4=pd.read_csv('Costomer+OrderForm.csv')
df4

Phillippines   Indonesia       Malta       India
0     Error Free  Error Free   Defective  Error Free
1     Error Free  Error Free  Error Free   Defective
2     Error Free   Defective   Defective  Error Free
3     Error Free  Error Free  Error Free  Error Free
4     Error Free  Error Free   Defective  Error Free
..           ...         ...         ...         ...
295   Error Free  Error Free  Error Free  Error Free
296   Error Free  Error Free  Error Free  Error Free
297   Error Free  Error Free   Defective  Error Free
298   Error Free  Error Free  Error Free  Error Free
299   Error Free   Defective   Defective  Error Free

[300 rows x 4 columns]

In [24]:
# information about the dataset
df4.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 300 entries, 0 to 299
Data columns (total 4 columns):
 #   Column        Non-Null Count  Dtype 
---  ------        --------------  ----- 
 0   Phillippines  300 non-null    object
 1   Indonesia     300 non-null    object
 2   Malta         300 non-null    object
 3   India         300 non-null    object
dtypes: object(4)
memory usage: 9.5+ KB


In [25]:
# statistical description about the dataset
df4.describe()

Phillippines   Indonesia       Malta       India
count           300         300         300         300
unique            2           2           2           2
top      Error Free  Error Free  Error Free  Error Free
freq            271         267         269         280

In [26]:
# to see counts of the values in dataset column
df4.Phillippines.value_counts()

Error Free    271
Defective      29
Name: Phillippines, dtype: int64

In [27]:
# to assign counts of the values from dataset column to array
Ph=df4.Phillippines.value_counts().values
Ph

array([271,  29])

In [28]:
# to assign counts of the values from dataset column to array
In=df4.Indonesia.value_counts().values
In

array([267,  33])

In [29]:
# to assign counts of the values from dataset column to array
Ma=df4.Malta.value_counts().values
Ma

array([269,  31])

In [30]:
# to assign counts of the values from dataset column to array
Ind=df4.India.value_counts().values
Ind

array([280,  20])

In [31]:
# making array of the all the different valuecounts
df4_array=np.array([Ph,In,Ma,Ind]).T
df4_array

array([[271, 267, 269, 280],
       [ 29,  33,  31,  20]])

In [32]:
# finding the p value by chi square contingency method on the array dataset
p_value4=chi2_contingency(df4_array)[1]
p_value4

0.2771020991233135

In [33]:
if p_value4 < 0.05:
    print("H0 is rejected and The defective % not varies by centre")
else:
    print("H0 is accepted and The defective % varies by centre")

H0 is accepted and The defective % varies by centre
